koji_wrapper/factory.py

```diff
@@ -33,6 +58,9 @@ def open_session(cls, profile=None, user_options=None,
         if debug:
             print({'user_options': user_options, 'session_options': options})
 
+        # TODO(jmls) Might be better to split this out slightly and either
```

Comment:
Agreed, I think an authenticate method of some kind would be good here, simply passing in authenticate and user_options.

Reply:
yea I think there might be another function hiding in here.

Reply:
done